### PR TITLE
366 add user guide surrogates

### PR DESCRIPTION
**Pull request by niklaswulkow**
Dominik's and Niklas' docu of surrogate models in bofire

### CODE REVIEW

**Review by R-M-Lee — COMMENTED** [review]
Great stuff!

Please add a line to mkdocs.yaml [here](https://github.com/experimental-design/bofire/blob/4f8eacbcfc4aa22481fccc961b83e84688859a6c/mkdocs.yaml#L12) to make this new page show up in the docs. I would put this under package architecture so you can just copy the bit used for Data Models vs Functional Components

**Comment by R-M-Lee** [comment]
@dlinzner-bcs I just reviewed this

**Comment by niklaswulkow** [comment]
Thank you, the line in mkdocs.yaml is now added.

> Great stuff!
> 
> Please add a line to mkdocs.yaml [here](https://github.com/experimental-design/bofire/blob/4f8eacbcfc4aa22481fccc961b83e84688859a6c/mkdocs.yaml#L12) to make this new page show up in the docs. I would put this under package architecture so you can just copy the bit used for Data Models vs Functional Components

**Comment by R-M-Lee** [comment]
ok. If you are happy with my suggestions you can just click on 'commit suggestion' at each appropriate block (you don't have to take my suggestion of course).

Then we can merge

File: docs/userguide_surrogates.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+Therefore, three criteria typically determine whether any candidate is selected as experimental proposal: the value of the surrogate model, the uncertainty of the model, and the acquisition function.
+
+## Surrogate model options
+BoFire offers the following classes of surrogate models.
+
+**Surrogate**|**Optimization of**|**When to use**|**Type**
+:-----:|:-----:|:-----:|:-----:
+[SingleTaskGPSurrogate](https://github.com/experimental-design/bofire/blob/main/bofire/surrogates/single\_task\_gp.py)|a single objective with real valued inputs|Limited data and black-box function is smooth|Gaussian process
+[RandomForestSurrogate](https://github.com/experimental-design/bofire/blob/main/bofire/surrogates/random\_forest.py)|a single objective|Rich data; black-box function does not have to be smooth|sklearn random forest implementation
```
[the remark this answers is not in the document]
Comment (niklaswulkow) [reply docs/userguide_surrogates.md:14]: Yes, I made an update and pushed

File: docs/userguide_surrogates.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+For the decision about which experiments to propose, one can additionally specify the acquisition function. This function determines if one rather wants to focus on exploitation, i.e., quickly approaching a close local optimum of the black-box function, or on exploration, i.e., exploring different regions of the input space first.
+
+Therefore, three criteria typically determine whether any candidate is selected as experimental proposal: the value of the surrogate model, the uncertainty of the model, and the acquisition function.
```
[the remark this answers is not in the document]
Comment (niklaswulkow) [reply docs/userguide_surrogates.md:6]: I would say acq. function *value* is the only quantity. But the function itself can be defined separately from the model prediction and uncertainty values and is then only applied to them.

File: docs/userguide_surrogates.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+[MixedSingleTaskGPSurrogate](https://github.com/experimental-design/bofire/blob/main/bofire/surrogates/mixed\_single\_task\_gp.py)|a single objective with categorical and real valued inputs|Limited data and black-box function is smooth|Gaussian process
+[XGBoostSurrogate](https://github.com/experimental-design/bofire/blob/main/bofire/surrogates/xgb.py)|a single objective|Rich data; black-box function does not have to be smooth|xgboost implementation of gradient boosting trees
+[TanimotoGP](https://github.com/experimental-design/bofire/blob/main/bofire/surrogates/tanimoto_gp.py)|a single objective|At least one input feature is a molecule represented as fingerprint|Gaussian process on a molecule space for which Tanimoto similarity determines the similarity between points|
+[MultiTaskGPSurrogate](https://github.com/experimental-design/bofire/blob/main/bofire/surrogates/single\_task\_gp.py)|a single objective with real valued inputs|Limited data and black-box function is smooth|Gaussian process
```
[the remark this answers is not in the document]
Comment (niklaswulkow) [reply docs/userguide_surrogates.md:19]: I couldn't find a MultiTaskGPSurrogate in the repo actually. I think we put it in although it doesn't exist yet. So I deleted it for now

**Pull request merged**